spec/cocoapods/generate/podfile_generator_spec.rb

```diff
@@ -613,6 +613,97 @@
       end
     end
 
+    context 'with external source pods specified' do
+      let(:config_options) { super().merge(external_source_pods: ['B' => [{ 'git' => 'https://github.com/pod.git' }]], use_podfile: false, use_lockfile: false, lockfile: nil, use_lockfile_versions: false) }
+
+      it 'generates the expected podfile' do
+        test = self
+        expected = Pod::Podfile.new do
+          self.defined_in_file = test.config.gen_dir_for_specs([Pod::Spec.new(nil, 'A')]).join('Podfile.yaml')
+
+          workspace 'A.xcworkspace'
+          project 'A.xcodeproj'
+
+          plugin 'cocoapods-disable-podfile-validations', 'no_abstract_only_pods' => true
+          plugin 'cocoapods-generate'
+
+          source 'https://cdn.cocoapods.org/'
+          source 'https://github.com/CocoaPods/Specs.git'
+          source 'https://github.com/Private/SpecsForks.git'
+
+          install! 'cocoapods',
+                   deterministic_uuids: false,
+                   disable_input_output_paths: false,
+                   generate_multiple_pod_projects: false,
+                   incremental_installation: false,
+                   share_schemes_for_development_pods: true,
+                   warn_for_multiple_pod_sources: false
+
+          use_frameworks!(true)
+
+          pod 'A', path: '../../Frameworks/A/A.podspec', testspecs: %w[Tests]
+
+          abstract_target 'Transitive Dependencies' do
+            pod 'B', git: 'https://github.com/pod.git'
+          end
+
+          target 'App-iOS' do
+          end
+          target 'App-macOS' do
+          end
+          target 'App-tvOS' do
+          end
+          target 'App-watchOS' do
+          end
+        end
+
+        expect(podfile_for_specs.to_yaml).to eq expected.to_yaml
+      end
+    end
+
+    context 'with external source pods specified as part of the ones being generated' do
```

Comment:
I don't totally understand the scenario this is covering. Mind adding a a bit of an explainer here?

Reply:
This particular tests ensures that _even if_ you specify the external source of the module you are generating workspace for it is ignored.

Like if I `be pod gen MyModule` and I set `MyModule` external source it should not take that into consideration.